minio/helpers.py

```diff
@@ -49,7 +49,24 @@
 _ALLOWED_HOSTNAME_REGEX = re.compile(
     '^((?!-)[A-Z\\d-]{1,63}(?<!-)\\.)*((?!-)[A-Z\\d-]{1,63}(?<!-))$',
     re.IGNORECASE)
+_EXTRACT_REGION_REGEX = re.compile('s3[.-]?(.+?).amazonaws.com')
```

Comment:
There is one more variant that seems easy to add here e.g. `s3.dualstack.us-east-2.amazonaws.com` - see https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region

Reply:
done